cts-java/src/test/java/org/ga4gh/cts/api/variants/VariantsPagingIT.java

```diff
@@ -0,0 +1,152 @@
+package org.ga4gh.cts.api.variants;
+
+import org.apache.avro.AvroRemoteException;
+import org.ga4gh.ctk.CtkLogs;
+import org.ga4gh.ctk.transport.URLMAPPING;
+import org.ga4gh.ctk.transport.protocols.Client;
+import org.ga4gh.cts.api.TestData;
+import org.ga4gh.cts.api.Utils;
+import org.ga4gh.methods.GAException;
+import org.ga4gh.methods.SearchVariantsRequest;
+import org.ga4gh.methods.SearchVariantsResponse;
+import org.ga4gh.models.Variant;
+import org.ga4gh.models.VariantSet;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Test the <tt>/variants/search</tt> paging behavior.
+ *
+ * @author Herb Jellinek
+ */
+@Category(VariantsTests.class)
+public class VariantsPagingIT implements CtkLogs {
+
+    private static Client client = new Client(URLMAPPING.getInstance());
+
+    /**
+     * Check that we can page 1 by 1 through the variants we receive from
+     * {@link org.ga4gh.ctk.transport.protocols.Client.Variants#searchVariants(SearchVariantsRequest)}.
+     *
+     * @throws AvroRemoteException if there's a communication problem or server exception ({@link GAException})
+     */
+    @Test
+    public void checkPagingOneByOneThroughCalls() throws AvroRemoteException {
```

Comment:
Please make sure that all the overlap corner cases are handled.
Things like:
  - overlap start only, end only
  - overlap middle
  - overlap exact match
  - overlap extends beyond
  - one base overlap
  - directly adjacent but non-overlapping